packages/web3-core/src/web3_context.ts

```diff
@@ -276,3 +285,7 @@ export type TransactionBuilder<
 	web3Context: Web3Context<API>;
 	privateKey?: HexString | Buffer;
 }) => Promise<ReturnType>;
+
+export abstract class Web3PluginBase<API extends Web3APISpec> extends Web3Context<API> {
```

Comment:
I assume this is how the rest of the modules in web3 are built. The only issue I see with this is once we have plugins that depend on other plugins, types won't work because we are augmenting Web3 class and plugins are using Web3Context.

Alternative of top of my head would be:
```
export const plugin1 = {
   namespace,
   init: (web3: Web3) => Plugin1Api
}

declare module web3{
  	interface Web3 {
		plugin1: Plugin1Api;
	}
}


```
and if you require plugin1 in plugin2:
```

import plugin2 from "plugin2"

export const plugin1 = {
   namespace,
   needs: [plugin1.namespace]
   // because we imported plugin2, Web3 class here will be augmented with plugin1 api
   init: (web3: Web3) => Plugin1Api
}
```

Reply:
#5490 tracks this